chatkit-core/src/test/kotlin/com/pusher/chatkit/RoomSubscriptionMemberFetchErrorFunctionalTest.kt

```diff
@@ -0,0 +1,58 @@
+package com.pusher.chatkit
+
+import com.google.common.truth.Truth.assertThat
+import com.pusher.chatkit.cursors.justConnectingCursorSubscription
+import com.pusher.chatkit.rooms.RoomEvent
+import com.pusher.chatkit.rooms.api.RoomMembershipApiType
+import com.pusher.chatkit.rooms.api.RoomReadStateApiType
+import com.pusher.chatkit.rooms.emptyRoomSubscription
+import com.pusher.chatkit.users.UserSubscriptionEvent
+import com.pusher.chatkit.users.usersFetchFailingWith
+import com.pusher.chatkit.util.FutureValue
+import elements.Errors
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object RoomSubscriptionMemberFetchErrorFunctionalTest : Spek({
+
+    val initialState = UserSubscriptionEvent.InitialState(
+            simpleUser("alice"),
+            listOf(simpleRoom("roomId1", "1")),
+            listOf(RoomReadStateApiType("roomId1", 0, null)),
+            listOf(RoomMembershipApiType("roomId1", listOf("alice", "bob")))
+    )
+
+    describe("given error when fetching joined room members") {
+        val networkError = Errors.network("test error")
+        val mockPlatformClient by memoized {
+            mockPlatformClient(
+                    userSubscription(initialState),
+                    emptyRoomSubscription(),
+                    justConnectingCursorSubscription(),
+                    usersFetchFailingWith(networkError)
+            )
+        }
+
+        val subject by memoized {
+            chatForFunctionalTest("alice", testPlatformClientFactory(mockPlatformClient))
+        }
+
+        describe("when room subscription is opened") {
+            lateinit var errorOccured: FutureValue<RoomEvent.ErrorOccurred>
+            beforeEachTest {
+                errorOccured = subject.subscribeRoomFor("1") { roomEvent ->
```

Comment:
Might be easier to read if the room name was `room1` or something more than 1 character on line 20 so that on line 43 it's easier to understand? This might work nicer as on line 21 you have the id being `roomId1` too.

Reply:
Good feedback! [Done](469c485faf66eb7fd5a00c7869f82c6080a613ae).